pkg/datapath/maps/map.go

```diff
@@ -95,9 +95,14 @@ func newMapSweeper(g endpointManager) *mapSweeper {
 // deleteMapIfStale uses the endpointManager implementation to determine for
 // the given path whether it should be deleted, and if so deletes the path.
 func (ms *mapSweeper) deleteMapIfStale(path string, filename string, endpointID string) {
-	if tmp, err := strconv.ParseUint(endpointID, 0, 16); err == nil {
+	if tmp, err := strconv.ParseUint(endpointID, 10, 16); err == nil {
 		epID := uint16(tmp)
-		if !ms.endpointExists(epID) {
+		if ms.endpointExists(epID) {
```

Comment:
Maybe it makes sense to add a comment saying that this branch of the if-statement is for removing an EP's map which name is in the old format?